background.ts

```diff
@@ -77,6 +78,11 @@ let enabledK8s: boolean;
 let pathManager: PathManager;
 const integrationManager: IntegrationManager = getIntegrationManager();
 let noModalDialogs = false;
+// Indicates whether the UI should be locked, settings changes should be disallowed
+// and possibly other things should be disallowed. As of the time of writing,
+// set to true when a snapshot is being created or restored.
+let locked = false;
```

Comment:
This name is too generic, especially for a global in this file. It should be something like `snapshotIsLocked`

Reply:
In [this comment](https://github.com/rancher-sandbox/rancher-desktop/issues/4236#issuecomment-1730777815) Jan mentioned a `locked` variable, specifically:

> I prefer a more generic name, in case we want to use the same mechanism for something else in the future.

I don't have a preference either way. @jandubois do you still think this?

Reply:
I don't want the APIs/variables for starting/stopping/locking the backend to be tied to snapshots, but I agree that a more specific name for a global variable is needed.

How about `backendIsLocked`?

A better name is welcome; it should somehow reflect that the VM is in a stopped state, and the user cannot start it again until it becomes unlocked.

Note that we'll need more than just a boolean; we need some kind of text that explains _why_ the VM is locked. For the next release we can hard-code this with the snapshot explanation, but eventually it must be settable by the component that requests the lock.

Reply:
`backendIsLocked` sounds good to me.

Setting its value to "" for when it is not locked and an explanation for when it is locked also sounds good. We can accept a boolean in the API, and just set `backendIsLocked` to `"Snapshot operation in progress"` when a `true` value is received (or when the lock file is present upon startup).